x/ibc/03-connection/types/codec.go

```diff
@@ -2,27 +2,51 @@ package types
 
 import (
 	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/ibc/03-connection/exported"
 )
 
-// SubModuleCdc defines the IBC connection codec.
-var SubModuleCdc *codec.Codec
-
-// RegisterCodec registers the IBC connection types
+// RegisterCodec registers the necessary x/ibc/03-connection interfaces and concrete types
+// on the provided Amino codec. These types are used for Amino JSON serialization.
 func RegisterCodec(cdc *codec.Codec) {
 	cdc.RegisterInterface((*exported.ConnectionI)(nil), nil)
 	cdc.RegisterInterface((*exported.CounterpartyI)(nil), nil)
-	cdc.RegisterConcrete(ConnectionEnd{}, "ibc/connection/ConnectionEnd", nil)
 
+	cdc.RegisterConcrete(ConnectionEnd{}, "ibc/connection/ConnectionEnd", nil)
 	cdc.RegisterConcrete(MsgConnectionOpenInit{}, "ibc/connection/MsgConnectionOpenInit", nil)
 	cdc.RegisterConcrete(MsgConnectionOpenTry{}, "ibc/connection/MsgConnectionOpenTry", nil)
 	cdc.RegisterConcrete(MsgConnectionOpenAck{}, "ibc/connection/MsgConnectionOpenAck", nil)
 	cdc.RegisterConcrete(MsgConnectionOpenConfirm{}, "ibc/connection/MsgConnectionOpenConfirm", nil)
+}
 
-	SetSubModuleCodec(cdc)
+// RegisterInterfaces register the ibc interfaces submodule implementations to protobuf
+// Any.
+func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	registry.RegisterImplementations((*sdk.Msg)(nil), &MsgConnectionOpenInit{})
+	registry.RegisterImplementations((*sdk.Msg)(nil), &MsgConnectionOpenTry{})
+	registry.RegisterImplementations((*sdk.Msg)(nil), &MsgConnectionOpenAck{})
+	registry.RegisterImplementations((*sdk.Msg)(nil), &MsgConnectionOpenConfirm{})
+	registry.RegisterInterface(
+		"cosmos_sdk.x.ibc.connection.v1.Connection",
+		(*exported.ConnectionI)(nil),
+		&ConnectionEnd{},
+	)
```

Comment:
Are there or are there intended to be other implementations of `ConnectionI` besides `ConnectionEnd`?